packages/language-server/src/ui5-model.ts

```diff
@@ -21,95 +16,82 @@ import {
 import { Fetcher } from "../api";
 import { getLogger } from "./logger";
 
-const DEFAULT_UI5_FRAMEWORK = "sapui5";
+const DEFAULT_UI5_FRAMEWORK = "SAPUI5";
 const DEFAULT_UI5_VERSION = "1.71.49";
 
 const UI5_FRAMEWORK_CDN_BASE_URL = {
-  openui5: "https://sdk.openui5.org/",
-  sapui5: "https://ui5.sap.com/",
+  OPENUI5: "https://sdk.openui5.org/",
+  SAPUI5: "https://ui5.sap.com/",
 };
 
+const semanticModelCache: Record<string, UI5SemanticModel> = Object.create(
+  null
+);
+function createSemanticModelCacheKey(
+  framework: UI5Framework,
+  version: string
+): string {
+  return `${framework}:${version}`;
+}
+
 export async function getSemanticModel(
```

Comment:
`getSemanticModel` reached 4 parameters.
is it time for a single `opts` param?

Reply:
could do so

Reply:
Looking into the follow-up tasks, I would keep it for now - not that I am against it, but there are many places to be adopted then... 😓